lightning/src/ln/channel.rs

```diff
@@ -3386,6 +3424,28 @@ impl<SP: Deref> ChannelContext<SP> where SP::Target: SignerProvider {
 		(context.holder_selected_channel_reserve_satoshis, context.counterparty_selected_channel_reserve_satoshis)
 	}
 
+	/// Check that a proposed channel value meets the channel reserve requirements or violates them (below reserve)
+	#[cfg(any(dual_funding, splicing))]
+	pub fn check_channel_value_meets_reserve_requirements(&self, proposed_channel_value: u64) -> Result<(), ChannelError> {
+		let holder_selected_channel_reserve_satoshis = get_v2_channel_reserve_satoshis(
+			proposed_channel_value, self.holder_dust_limit_satoshis);
+		if proposed_channel_value < holder_selected_channel_reserve_satoshis {
+			return Err(ChannelError::Warn(format!(
+				"Proposed channel value below reserve mandated by holder, {} vs {}",
+				proposed_channel_value, holder_selected_channel_reserve_satoshis,
+			)));
+		}
+		let counterparty_selected_channel_reserve_satoshis = get_v2_channel_reserve_satoshis(
+			proposed_channel_value, self.counterparty_dust_limit_satoshis);
+		if proposed_channel_value < counterparty_selected_channel_reserve_satoshis {
+			return Err(ChannelError::Warn(format!(
+				"Proposed channel value below reserve mandated by counterparty, {} vs {}",
+				proposed_channel_value, counterparty_selected_channel_reserve_satoshis,
+			)));
+		}
```

Comment:
Not quite sure I understand these checks. Don't we want to make sure the balance (not the total value) doesn't violate the reserve requirements?